### ISSUE TITLE
Limited number of Reflection Probes 

### ISSUE THREAD

**Issue by zoiba** [godotengine/godot#36399]
Godot 3.2

verified on PC Linux and Oculus Quest

I'm using GLES2, I've set multiple reflection probes for covering the various rooms in a building, only the first two probes works.

Hide all the probes and the show them by clicking in a different sequence, only the first two you made visible will work.

[testcase_reflection_probe.zip](https://github.com/godotengine/godot/files/4233166/testcase_reflection_probe.zip)

**Comment by clayjohn** [comment]
Does the building happen to be one large mesh? Only two probes can be used per mesh.

**Comment by zoiba** [comment]
yes it does, large meshes are involved in building design. haven't found such limitation written on the documentation, thank you for the information.

**Comment by clayjohn** [comment]
Oops. Looks like I misunderstood what you needed. 

Only two probes can be visible at a single overlapping point per-mesh. However, you can have more per-mesh as long as they don't overlap. The renderer should select the closest two to use as you move around the building. It works correctly in GLES3.

**Comment by zoiba** [comment]
I reworked the test case project so to don't have overlapping reflection probes. If I unhide the first probe it works fine. The second probe works but it seems the "enable interior" turns on and off depending form camera orientation. The third and fourth probes simply don't work. If I Unhide the fifth probe Godot freezes immediately or after moving the camera.
Here the test case updated:
[testcase_reflection_probe.zip](https://github.com/godotengine/godot/files/4235500/testcase_reflection_probe.zip)

**Comment by KoBeWi** [comment]
Can anyone still reproduce this bug in Godot 3.2.3 or any later release?

**Comment by clktmr** [comment]
I can reproduce this bug in 3.3.4. Behaviour is exactly as OP has written. A single probe works fine. Two probes cause flickering of ambient light depending on camera angle. A third and fourth probe don't render at all. I played around with the reflection settings in Project/Render Quality, but couldn't see any difference. Only happens with the GLES2 backend, GLES3 works fine.
